typing_tests/pine-typed-result.ts

```diff
@@ -167,31 +174,35 @@ export let aStringOrUndefined: string | undefined;
 // $expand w/o $select
 
 {
-	type deviceOptionsExpandNavigationResourceString = PineClient.TypedResult<
-		BalenaSdk.Device,
-		{
-			$expand: 'belongs_to__application';
-		}
+	type deviceOptionsExpandNavigationResourceString = NonNullable<
+		BalenaSdk.Pine.OptionsToResponse<
+			BalenaSdk.Device['Read'],
+			{
+				$expand: ['is_of__device_type'];
+			},
+			number
+		>
 	>;
 
 	const result: deviceOptionsExpandNavigationResourceString = {} as any;
 
-	aNumber = result.belongs_to__application[0].id;
+	aNumber = result.is_of__device_type[0].id;
 	aNumber = result.id;
 	aString = result.device_name;
-
-	// @ts-expect-error test case
-	aAny = result.belongs_to__application[1];
+	aAny = result.is_of__device_type[1];
```

Comment:
```suggestion
	// TODO: TS should complain about is_of__device_type[1] since the prop should ideally be a single item tuple
	aNumber = result.is_of__device_type[1].id;
```